src/containers/createRoom.js

```diff
@@ -9,35 +9,91 @@ export class createRoom extends Component {
     username: "",
   };
 
-  handleChange = (event) => {
+  handleChange(event) {
     this.setState({ [event.target.name]: event.target.value });
   };
 
-  handleSubmit = (event) => {
-    event.preventDefault();
+  getRooms() {
     const reqObj = {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
         Accept: "application/json",
       },
-      body: JSON.stringify({ room: this.state }),
+      body: JSON.stringify({
+        room: this.state
+      }),
     };
-    fetch("http://localhost:3000/rooms", reqObj)
-      .then((resp) => resp.json())
-      .then((resp) => {
-        localStorage.setItem("token", resp.jwt);
-        this.props.setCreateRoom(
-          resp.user,
-          resp.room.room_name,
-          resp.room.host_id,
-          resp.room.host_name
-        );
-        this.props.history.push(`/room/${resp.room.id}`);
-      });
+
+    return fetch("http://localhost:3000/rooms", reqObj)
+      .then((resp) => resp.json());
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    this.getRooms().then((response) => {
+      let { jwt, user, room} = response;
+      
+      localStorage.setItem("token", jwt);
+
+      this.props.setCreateRoom(
+        user,
+        room.room_name,
+        room.host_id,
+        room.host_name
+      );
+      
+      this.props.history.push(`/room/${room.id}`);
+    });
+
     this.setState({ room_name: "", password: "", username: "" });
   };
 
+  renderForm() {
```

Comment:
You'll want to develop the skill of recognizing conceptual areas of a page to extract as components. You can often find boundaries between layout HTML elements (like the bootstrap `Row` and `Col` components) and your app's actual visual components (like forms or profile avatars).

Here, I'd extract the form to make it clearer what's what between the layout and the longer form code.